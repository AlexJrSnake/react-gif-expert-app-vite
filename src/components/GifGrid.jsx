import React, {useEffect, useState} from 'react'
// import {getGif} from '../helpers/getGifs'
import {GifItem} from '../components/GifItem'
import {useFetchGifs} from "../hooks/useFetchGifs"
import PropTypes from "prop-types";

export const GifGrid = ({category}) => {

    const { imagenes, isLoading} = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>

            {
                isLoading && <h2>Cargando...</h2>
            }

            <div className="card-grid">
                { 
                    imagenes.map( (element, index) => {
                        return  <GifItem key={element.id} 
                                         {...element}> 
                                </GifItem>
                    }) 
                }            
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}