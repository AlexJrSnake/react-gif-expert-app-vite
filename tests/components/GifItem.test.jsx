import { fireEvent, render, screen } from "@testing-library/react"
import {GifItem} from "../../src/components/GifItem"

describe('<GifItem></GifItem>', () => { 

    const title = 'Prueba';
    const url = 'https://www.tumgir.com/tag/Julia%20Katsushika'

    test('snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot()
     })


     test('debe mostrar la imagen con el URL indicado', () => {
        render(<GifItem title={title} url={url}/>)
        // screen.debug();
        const { src , alt } = screen.getByRole('img')
        expect( src ).toBe(url);
        expect( alt ).toBe(title);
     })

     test('debe mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url}/>)
        expect( screen.getByText(title) ).toBeTruthy();
     })

})