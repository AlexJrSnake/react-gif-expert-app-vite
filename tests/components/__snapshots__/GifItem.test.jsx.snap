// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem></GifItem> snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Prueba"
      src="https://www.tumgir.com/tag/Julia%20Katsushika"
    />
    <p>
      Prueba
    </p>
  </div>
</div>
`;
